### PR TITLE
make resizeTimeMs metric as a Gauge instead of a Meter

### PR DESCRIPTION
**Pull request by mqliang**
## Description

As discussed in https://github.com/apache/incubator-pinot/pull/6680#discussion_r595643785_ :

resizeTimeMs should not be Meter, since Meter has counters, whereas Gauge has the absolute values. This PR moves resizeTimeMs metric to Gauge and set the unit as "milliseconds"

## Upgrade Notes
Does this PR prevent a zero down-time upgrade? (Assume upgrade order: Controller, Broker, Server, Minion)
* [ ] Yes (Please label as **<code>backward-incompat</code>**, and complete the section below on Release Notes)

Does this PR fix a zero-downtime upgrade introduced earlier?
* [ ] Yes (Please label this as **<code>backward-incompat</code>**, and complete the section below on Release Notes)

Does this PR otherwise need attention when creating release notes? Things to consider:
- New configuration options
- Deprecation of configurations
- Signature changes to public methods/interfaces
- New plugins added or old plugins removed
* [ ] Yes (Please label this PR as **<code>release-notes</code>** and complete the section on Release Notes)
## Release Notes
If you have tagged this as either backward-incompat or release-notes,
you MUST add text here that you would like to see appear in release notes of the
next release.

If you have a series of commits adding or enabling a feature, then
add this section only in final commit that marks the feature completed.
Refer to earlier release notes to see examples of text

## Documentation
If you have introduced a new feature or configuration, please add it to the documentation as well.
See https://docs.pinot.apache.org/developers/developers-and-contributors/update-document

### CODE REVIEW

File: pinot-common/src/main/java/org/apache/pinot/common/metrics/BrokerMeter.java [hunk 1]
```diff
@@ -64,8 +64,7 @@
   ENTRIES_SCANNED_IN_FILTER("documents", false),
   ENTRIES_SCANNED_POST_FILTER("documents", false),
 
-  NUM_RESIZES("resizes", false),
-  RESIZE_TIME_MS("resizeTimeMs", false),
+  NUM_RESIZES("numResizes", false),
```
Comment (mqliang) [inline pinot-common/src/main/java/org/apache/pinot/common/metrics/BrokerMeter.java:67]: @siddharthteotia I also change the unit of BrokerMeter from "resizes" to "numResizes" to keep consistence with ServerMeter. Let me know if I need change back.

**Review by siddharthteotia — APPROVED** [review]
Thanks for fixing this.

**Comment by mqliang** [comment]
unit test failed due to flaky networking:
```
Error:  Failed to execute goal on project pinot-kafka-0.9: Could not resolve dependencies for project org.apache.pinot:pinot-kafka-0.9:jar:0.7.0-SNAPSHOT: Failed to collect dependencies at org.apache.kafka:kafka_2.10:jar:0.9.0.1: Failed to read artifact descriptor for org.apache.kafka:kafka_2.10:jar:0.9.0.1: Could not transfer artifact org.apache.kafka:kafka_2.10:pom:0.9.0.1 from/to central (https://repo.maven.apache.org/maven2): Transfer failed for https://repo.maven.apache.org/maven2/org/apache/kafka/kafka_2.10/0.9.0.1/kafka_2.10-0.9.0.1.pom: Connection reset -> [Help 1]
```

close&re-open to trigger a re-run.

File: pinot-common/src/main/java/org/apache/pinot/common/metrics/ServerGauge.java [hunk 2]
```diff
@@ -41,6 +41,7 @@
   REALTIME_OFFHEAP_MEMORY_USED("bytes", false),
   REALTIME_SEGMENT_NUM_PARTITIONS("realtimeSegmentNumPartitions", false),
   LLC_SIMULTANEOUS_SEGMENT_BUILDS("llcSimultaneousSegmentBuilds", true),
+  RESIZE_TIME_MS("resizeTimeMs", false),
```
Comment (siddharthteotia) [inline pinot-common/src/main/java/org/apache/pinot/common/metrics/ServerGauge.java:44]: This should be defined as RESIZE_TIME_MS("milliSeconds", false). See the other duration related gauge metrics.

Comment (siddharthteotia) [reply pinot-common/src/main/java/org/apache/pinot/common/metrics/ServerGauge.java:44]: It actually may not matter but if you look at the constructor of ServerGauge, the unit is the first argument

File: pinot-common/src/main/java/org/apache/pinot/common/metrics/BrokerGauge.java [hunk 3]
```diff
@@ -30,7 +30,8 @@
   MAX_BURST_QPS("tables", false),
   QUERY_RATE_LIMIT_DISABLED("queryQuota", true),
   NETTY_CONNECTION_CONNECT_TIME_MS("nettyConnection", true),
-  REQUEST_SIZE("requestSize", false);
+  REQUEST_SIZE("requestSize", false),
+  RESIZE_TIME_MS("resizeTimeMs", false);
```
Comment (siddharthteotia) [inline pinot-common/src/main/java/org/apache/pinot/common/metrics/BrokerGauge.java:34]: I think this should be RESIZE_TIME_MS("milliseconds", false)

**Review by siddharthteotia — APPROVED** [review]
Thanks for fixing this.

**Pull request merged**